src/main/common/utils.js

```diff
@@ -188,6 +188,78 @@ gpii.app.isPointInRect = function (point, rectangle) {
            rectangle.y <= point.y && point.y <= rectangle.y + rectangle.height;
 };
 
+/**
+ * Checks if the buttonList attribute exists in the siteConfig object
+ * @param {Object} siteConfig - instance of the siteConfig object
+ * @returns {Boolean}
+ */
+gpii.app.isButtonList = function(siteConfig) {
+    return fluid.isValue(siteConfig.buttonList);
+};
+
+/**
+ * Looks for a `id` and matches it to the provided string
+ * return empty array when there is no button found
+ * @param {String} buttonId - the `id` of the button
+ * @param {Array} availableButtons - the full settings list
+ * @returns {Array}
+ */
+gpii.app.findButtonId = function(buttonId, availableButtons) {
+    var returnData = [];  // returning empty array by default
```

Comment:
Why an empty array when you return a single button if it is found?

Reply:
Yeah, no sense to return empty array, returning false now.